### PR TITLE
Start the process of renaming platform to CxPlat

### PR DESCRIPTION
**Pull request by ThadHouse**
Currently, only actual functions have been changed. Macros and globals still need to be done.

### CODE REVIEW

File: src/core/crypto.c [hunk 1]
```diff
@@ -182,7 +182,7 @@ QuicCryptoInitialize(
     }
 
     Status =
-        QuicPacketKeyCreateInitial(
+        CxPlatPacketKeyCreateInitial(
```
Comment (nibanks) [inline src/core/crypto.c:185]: This is a special case that might need to be Quic*

File: src/inc/quic_crypt.h [hunk 2]
```diff
@@ -84,7 +84,7 @@ QuicKeyLength(
     case QUIC_AEAD_AES_256_GCM:
```
Comment (nibanks) [inline src/inc/quic_crypt.h:84]: QUIC_AEAD needs to be updated.

File: src/inc/quic_crypt.h [hunk 3]
```diff
@@ -121,7 +121,7 @@ QuicHashLength(
     case QUIC_HASH_SHA384: return 48;
```
Comment (nibanks) [inline src/inc/quic_crypt.h:121]: QUIC_HASH needs to be updated.

File: src/inc/quic_datapath.h [hunk 4]
```diff
@@ -196,15 +196,15 @@ typedef struct QUIC_RECV_DATA {
 // Gets the corresponding receive data from its context pointer.
 //
 QUIC_RECV_DATA*
-QuicDataPathRecvPacketToRecvData(
+CxPlatDataPathRecvPacketToRecvData(
     _In_ const QUIC_RECV_PACKET* const RecvPacket
```
Comment (nibanks) [inline src/inc/quic_datapath.h:200]: All the QUIC_* types in here can be changed.

File: src/inc/quic_platform.h [hunk 5]
```diff
@@ -44,9 +44,9 @@ typedef struct QUIC_LIST_ENTRY {
     struct QUIC_LIST_ENTRY* Blink;
 } QUIC_LIST_ENTRY;
 
-typedef struct QUIC_SINGLE_LIST_ENTRY {
-    struct QUIC_SINGLE_LIST_ENTRY* Next;
-} QUIC_SINGLE_LIST_ENTRY;
+typedef struct CXPLAT_SINGLE_LIST_ENTRY {
+    struct CXPLAT_SINGLE_LIST_ENTRY* Next;
+} CXPLAT_SINGLE_LIST_ENTRY;
```
Comment (nibanks) [inline src/inc/quic_platform.h:49]: Let's rename to _SLIST_

File: src/manifest/MsQuicEtw.man [hunk 6]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version='1.0' encoding='utf-8' standalone='yes'?>
+<?xml version='1.0' encoding='utf-8' standalone='yes'?>
```
Comment (nibanks) [inline src/manifest/MsQuicEtw.man:1]: What changed here?

File: src/inc/quic_trace.h [hunk 7]
```diff
@@ -195,30 +195,30 @@ QuicEtwCallback(
 
 inline
 void
-QuicTraceStubVarArgs(
+CxPlatTraceStubVarArgs(
```
Comment (nibanks) [inline src/inc/quic_trace.h:198]: I think we should keep al this as Quic

File: src/inc/quic_toeplitz.h [hunk 8]
```diff
@@ -52,15 +52,15 @@ typedef struct QUIC_TOEPLITZ_HASH {
 // Initializes the Toeplitz hash structure. Toeplitz->HashKey must be set first.
 //
 void
-QuicToeplitzHashInitialize(
+CxPlatToeplitzHashInitialize(
     _Inout_ QUIC_TOEPLITZ_HASH* Toeplitz
```
Comment (nibanks) [inline src/inc/quic_toeplitz.h:56]: Rename struct too

File: src/manifest/MsQuicEtw.man (hunk 6 above)
Comment (ThadHouse) [reply src/manifest/MsQuicEtw.man:1]: Likely just a BOM issue. I'll fix it.

File: src/inc/quic_crypt.h (hunk 2 above)
Comment (ThadHouse) [reply src/inc/quic_crypt.h:84]: Do they? This and QUIC_HASH seem very QUIC specific.

Comment (nibanks) [reply src/inc/quic_crypt.h:84]: Only QUIC_PACKET stuff is QUIC specific. Otherwise they are all fairly generic crypto abstractions.

File: src/core/ack_tracker.h [hunk 9]
```diff
@@ -100,7 +100,7 @@ QuicAckTrackerAckPacket(
     _Inout_ QUIC_ACK_TRACKER* Tracker,
     _In_ uint64_t PacketNumber,
     _In_ uint64_t RecvTimeUs,
-    _In_ QUIC_ECN_TYPE ECN,
+    _In_ CXPLAT_ECN_TYPE ECN,
```
Comment (ThadHouse) [inline src/core/ack_tracker.h:103]: Is ECN a quic specific thing, or is that applicable to tcp/udp?

Comment (nibanks) [reply src/core/ack_tracker.h:103]: It is **IP** specific, so not QUIC specific.

**Comment by ThadHouse** [comment]
I think this is at a good point to be merged in. This is NOT a PR we want to go stale, so pulling this in in chunks is likely the right decision. We can then work out some of the other circular issues.

File: src/tools/etw/worker.c [hunk 10]
```diff
@@ -63,7 +63,7 @@ NewWorker(
     _In_ PEVENT_RECORD ev
     )
 {
-    QUIC_EVENT_DATA_WORKER* EvData = (QUIC_EVENT_DATA_WORKER*)ev->UserData;
+    CXPLAT_EVENT_DATA_WORKER* EvData = (CXPLAT_EVENT_DATA_WORKER*)ev->UserData;
```
Comment (nibanks) [inline src/tools/etw/worker.c:66]: Please undo these changes.

File: src/platform/storage_winkernel.c [hunk 11]
```diff
@@ -86,8 +86,8 @@ QuicStorageRegKeyChangeCallback(
 typedef struct QUIC_STORAGE {
```
Comment (nibanks) [inline src/platform/storage_winkernel.c:86]: QUIC_STORAGE needs to be updated please

File: src/platform/platform_winkernel.c [hunk 12]
```diff
@@ -57,13 +57,13 @@ typedef struct _SYSTEM_BASIC_INFORMATION {
 
 
 uint64_t QuicPlatformPerfFreq;
```
Comment (nibanks) [inline src/platform/platform_winkernel.c:59]: `QuicPlatformPerfFreq` needs to be updated

**Review by nibanks — APPROVED** [review]
